roles/matrix-synapse/templates/synapse/homeserver.yaml.j2

```diff
@@ -1039,6 +1039,27 @@ media_store_path: "/matrix-media-store-parent/{{ matrix_synapse_media_store_dire
 #    config:
 #       directory: /mnt/some/other/directory
 
+# Matrix S3 Storage Provider
+{% if matrix_synapse_s3_media_store_enabled %}
+media_storage_providers:
+- module: s3_storage_provider.S3StorageProviderBackend
+  # In the context of this module the local store means the S3 Bucket
+  # and the remote store the local volume ¯\_(ツ)_/¯ (i swaped it for convenience)
+  store_local: {{ matrix_synapse_s3_media_store_remote }}
+  store_remote: {{ matrix_synapse_s3_media_store_local }}
+  store_synchronous: {{ matrix_synapse_s3_media_store_sync }}
+  config:
+    bucket: {{ matrix_synapse_s3_media_store_bucket }}
+    # All of the below options are optional, for use with non-AWS S3-like
+    # services, or to specify access tokens here instead of some external method.
+    region_name: {{ matrix_synapse_s3_media_store_region }}
+    endpoint_url: {{ matrix_synapse_s3_media_store_endpoint }}
+    access_key_id: {{ matrix_synapse_s3_media_store_access }}
+    secret_access_key: {{ matrix_synapse_s3_media_store_key }}
```

Comment:
Using [`|to_json`](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#formatting-data-yaml-and-json) for all of these is safer.

```diff
-secret_access_key: {{ matrix_synapse_s3_media_store_key }}
+secret_access_key: {{ matrix_synapse_media_storage_provider_s3_secret_access_key|to_json }}
```

Reply:
Added the `|toJson` attribute to all String based defined input variables.